src/types/base.ts

```diff
@@ -80,6 +81,11 @@ export abstract class ZodType<Type, Def extends ZodTypeDef = ZodTypeDef> {
     return this;
   };
 
+  error(customError: object) {
```

Comment:
I think it should also accept `customError` as `string`

Reply:
added functionality and test